hal/src/rtl872x/pwm_hal.cpp

```diff
@@ -129,11 +123,13 @@ int pwmTimBaseInit(rtl_pwm_info_t* info) {
     RCC_PeriphClockCmd(APBPeriph_GTIMER, APBPeriph_GTIMER_CLOCK, ENABLE);
 
     if (info->state == PWM_STATE_UNKNOWN) {
-        memset(info->pins, PIN_INVALID, sizeof(info->pins));
-        memset(info->resolution, DEFAULT_RESOLUTION, sizeof(info->resolution));
-        // Juts in case
+        for( int i = 0; i < MAX_PWM_CHANNEL_NUM; i++){
+            info->channels[i].pin = PIN_INVALID;
+            info->channels[i].resolution = DEFAULT_RESOLUTION;
+            info->channels[i].state = PWM_STATE_DISABLED;
+        }
+        // Just in case
         RTIM_Cmd(info->tim, DISABLE);
-        info->state = PWM_STATE_DISABLED;

```

Comment:
I recall that updating the state here right after the PWM is disabled is just to avoid potential race condition. But yeah, it doesn't hurt even if we don't update the state herein.